internal/consul/connection.go

```diff
@@ -81,6 +83,46 @@ func (c *client) Wait(until time.Duration) error {
 }
 
 func (c *client) WatchServers(ctx context.Context) error {
+	if !c.config.UseDynamic {
+		cfg := c.config.ApiClientConfig
+		cfg.Address = fmt.Sprintf("%s:%d", c.config.Addresses, c.config.HTTPPort)
+
+		var err error
+		var client *api.Client
+		var token string
+		if c.config.Credentials.Type == discovery.CredentialsTypeLogin {
+			client, err = api.NewClient(cfg)
+			if err != nil {
+				c.initialized <- err
+				return err
+			}
+			client, token, err = login(ctx, client, cfg, c.config)
+			if err != nil {
+				c.initialized <- err
+				return err
+			}
+			defer logout(client, token, c.config)
+		} else {
+			// this might be empty
+			cfg.Token = c.config.Credentials.Static.Token
+			client, err = api.NewClient(cfg)
+			if err != nil {
+				c.initialized <- err
+				return err
+			}
+		}
+
+		c.mutex.Lock()
+		c.client = client
+		c.token = cfg.Token
```

Comment:
⚠️  I think this should get set to `token` (and set `token = c.config.Credentials.Static.Token; cfg.Token = token` in the non-login path) unless `login` is mutating `cfg.Token`?

**Edit:** reading the `login` impl it looks like it does mutate `cfg.Token` so this should be okay, but it is a bit unintuitive.